crates/storage/provider/src/test_utils/mod.rs

```diff
@@ -26,7 +26,8 @@ pub fn create_test_provider_factory() -> ProviderFactory<Arc<TempDatabase<Databa
 pub fn create_test_provider_factory_with_chain_spec(
     chain_spec: Arc<ChainSpec>,
 ) -> ProviderFactory<Arc<TempDatabase<DatabaseEnv>>> {
+    let (static_dir, _) = create_test_static_files_dir();
     let db = create_test_rw_db();
-    ProviderFactory::new(db, chain_spec, create_test_static_files_dir())
+    ProviderFactory::new(db, chain_spec, static_dir.into_path())
```

Comment:
@shekhirin Tests running with this methods still can not delete the `temp_dir` by itself, Would it be beneficial to create a class, perhaps named TestProviderFactory, that owns a temp_dir? This way, when the provider is dropped, the temp_dir would also be deleted.

Reply:
Good suggestion, I will apply the required change in tomorrow.